gcloud/bigquery/job.py

```diff
@@ -0,0 +1,702 @@
+
+    @property
+    def input_files(self):
+        """Count of source files.
+
+        :rtype: integer, or ``NoneType``
+        :returns: the count (None until set from the server).
+        """
+        statistics = self._properties.get('statistics')
+        if statistics is not None:
+            return int(statistics['load']['inputFiles'])
+
+    @property
+    def output_bytes(self):
+        """Count of bytes saved to destination table.
+
+        :rtype: integer, or ``NoneType``
+        :returns: the count (None until set from the server).
+        """
+        statistics = self._properties.get('statistics')
+        if statistics is not None:
+            return int(statistics['load']['outputBytes'])
+
+    @property
+    def output_rows(self):
+        """Count of rows saved to destination table.
+
+        :rtype: integer, or ``NoneType``
+        :returns: the count (None until set from the server).
+        """
+        statistics = self._properties.get('statistics')
+        if statistics is not None:
+            return int(statistics['load']['outputRows'])
+
+    @property
+    def error_result(self):
+        """Error information about the job as a whole.
+
+        :rtype: mapping, or ``NoneType``
+        :returns: the error information (None until set from the server).
+        """
+        status = self._properties.get('status')
+        if status is not None:
+            return status['errorResult']
+
+    @property
+    def errors(self):
+        """Information about individual errors generated by the job.
+
+        :rtype: list of mappings, or ``NoneType``
+        :returns: the error information (None until set from the server).
+        """
+        status = self._properties.get('status')
+        if status is not None:
+            return status['errors']
+
+    @property
+    def state(self):
+        """Status of the job.
+
+        :rtype: string, or ``NoneType``
+        :returns: the state (None until set from the server).
+        """
+        status = self._properties.get('status')
+        if status is not None:
+            return status['state']
+
+    @property
+    def allow_jagged_rows(self):
+        """Allow rows with missing trailing commas for optional fields.
+
+        :rtype: boolean, or ``NoneType``
+        :returns: The value as set by the user, or None (the default).
+        """
+        return self._configuration._allow_jagged_rows
+
+    @allow_jagged_rows.setter
+    def allow_jagged_rows(self, value):
+        """Update allow_jagged_rows.
+
+        :type value: boolean
+        :param value: new allow_jagged_rows
+
+        :raises: ValueError for invalid value types.
+        """
+        if not isinstance(value, bool):
+            raise ValueError("Pass a boolean")
+        self._configuration._allow_jagged_rows = value
+
+    @allow_jagged_rows.deleter
+    def allow_jagged_rows(self):
+        """Delete allow_jagged_rows."""
+        del self._configuration._allow_jagged_rows
```

Comment:
Wouldn't we rather have it be `None` rather than deleted? Empirically it seems to fall back to class value:

```python
>>> class A(object):
...     b = None
... 
>>> a = A()
>>> print(a.b)
None
>>> a.b = 200
>>> print(a.b)
200
>>> del a.b
>>> print(a.b)
None
```